recipe/meta.yaml

```
@@ -1,33 +1,78 @@
 {% set name = "scikit-optimize" %}
 {% set version = "0.9.0" %}
-{% set sha256 = "77d8c9e64947fc9f5cc05bbc6aed7b8a9907871ae26fe11997fd67be90f26008" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  url: https://github.com/{{ name }}/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: b5602b22c7aacabe3afd36619f662d1f287f2e3fe9a5aa19bbc83f851b30b7bf
+  patches:
+    # this patch fixes the use of deprecated np.int (numpy >=1.20)
+    - 0001-replace-deprecated-np.int.patch
+    # this patch fixes an incorrectly chosen error rounding criteria used in the required scikit-learn version
+    - 0002-use-compatible-mse-criteria-with-scikit-learn.patch
 
 build:
   number: 1
-  script: {{ PYTHON }} -m pip install . -vv
-  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
+  build:
+    - patch  # [not win]
+    - m2-patch  # [win] 
   host:
-    - python >=3.6
+    - python
     - pip
+    - wheel
+    - setuptools
   run:
-    - python >=3.6
-    - numpy
-    - scipy >=0.14.0
-    - scikit-learn >=0.19.1
-    - matplotlib-base
-    - pyaml
-    - joblib
+    - python
+    - {{ pin_compatible('numpy') }}
```

```suggestion
    - numpy >=1.13.3
```
pin_compatible is only effective when the dependency is also used in host. Here the result is likely an unpinned numpy.
Here numpy is only used as a python dependency, therefore we should follow upstream requirements.
https://github.com/scikit-optimize/scikit-optimize/blob/v0.9.0/setup.py#L45

---

Good to know, thank you!